### DISCUSSION TITLE
Using Pinia In OnBeforeRender

### DISCUSSION THREAD

**Discussion by hamedg68** [vikejs/vike#757] [Discussion]
According to **Pinia example** I can not use Pinia in **onBeforeRender** hook when server runs at first, because it has not activated yet, but by reloading the page it works.

Error message : 
```_Error: [🍍]: getActivePinia was called with no active Pinia. Did you forget to install Pinia?_```

I know the solution is using **onServerPrefetch** but if I use this option I miss some **VPS** features for instance set title for each page which is fetched from server.

Is there any way to solve this issue, I mean when serve runs, at first loading, **Pinia** get activated in  **onBeforeRender** ?

**Comment by brillout** [comment]
Have you found a solution? (PR welcome to https://vite-plugin-ssr.com/pinia btw.)

**Reply by hamedg68** [reply]
No, I don't have any idea

**Reply by brillout** [reply]
Ok. Let me know if you believe there is anything VPS can do to unblock the situation.

Keep us updated.
